core/src/main/scala/io/aiven/guardian/kafka/KafkaClient.scala

```diff
@@ -14,44 +15,58 @@ import akka.stream.scaladsl.SourceWithContext
 import com.typesafe.scalalogging.StrictLogging
 import io.aiven.guardian.kafka.configs.KafkaCluster
 import io.aiven.guardian.kafka.models.ReducedConsumerRecord
+import org.apache.kafka.clients.consumer.ConsumerConfig
 import org.apache.kafka.common.serialization.ByteArrayDeserializer
 
+import scala.collection.immutable
 import scala.concurrent.Future
 
 import java.util.Base64
 
 /** A Kafka Client that uses Alpakka Kafka Consumer under the hood to create a stream of events from a Kafka cluster. To
   * configure the Alpakka Kafka Consumer use the standard typesafe configuration i.e. akka.kafka.consumer (note that the
   * `keySerializer` and `valueSerializer` are hardcoded so you cannot override this).
-  * @param configure
+  * @param configureConsumer
   *   A way to configure the underlying Kafka consumer settings
+  * @param configureCommitter
+  *   A way to configure the underlying kafka committer settings
   * @param system
   *   A classic `ActorSystem`
   * @param kafkaClusterConfig
   *   Additional cluster configuration that is needed
   */
 class KafkaClient(
-    configure: Option[ConsumerSettings[Array[Byte], Array[Byte]] => ConsumerSettings[Array[Byte], Array[Byte]]] = None
+    configureConsumer: Option[
+      ConsumerSettings[Array[Byte], Array[Byte]] => ConsumerSettings[Array[Byte], Array[Byte]]
+    ] = None,
+    configureCommitter: Option[
+      CommitterSettings => CommitterSettings
+    ] = None
 )(implicit system: ActorSystem, kafkaClusterConfig: KafkaCluster)
     extends KafkaClientInterface
     with StrictLogging {
-  override type CursorContext = Committable
-  override type Control       = Consumer.Control
+  override type CursorContext        = CommittableOffset
+  override type Control              = Consumer.Control
+  override type BatchedCursorContext = CommittableOffsetBatch
 
   if (kafkaClusterConfig.topics.isEmpty)
     logger.warn("Kafka Cluster configuration has no topics set")
 
   private[kafka] val consumerSettings = {
     val base = ConsumerSettings(system, new ByteArrayDeserializer, new ByteArrayDeserializer)
-    configure.fold(base)(block => block(base))
+    configureConsumer
+      .fold(base)(block => block(base))
+      .withProperties(
+        ConsumerConfig.AUTO_OFFSET_RESET_CONFIG -> "earliest"
```

Comment:
This is the magic configuration without which the suspend/resume case does not work.